spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/mail/MailSenderAutoConfiguration.java

```diff
@@ -40,6 +41,7 @@
  */
 @Configuration
 @ConditionalOnClass({MailSender.class, AmazonSimpleEmailService.class})
+@ConditionalOnMissingBean(MailSender.class)
```

Comment:
Oh! Didn't notice that. 
Right, that would be better.
`org.springframework.boot.autoconfigure.mail.MailSenderAutoConfiguration` has the same `@ConditionalOnMissingBean(MailSender.class)`, so making it `after` would collide with SES independently of the `spring.mail.*` properties. 
With what you suggest we can still have SES if we don't use `spring.mail.*` properties, and can override SES, if we would like to use `spring.mail.*` and still rely just on the default Spring Boot `MailSender`.
In case of both we don't have choice unless configure everything manually. Because `@ConditionalOnMissingBean(MailSender.class)` kills both auto-configuration, when we have even only just one custom `MailSender`.

Am I correct?